framework/src/base/MooseVariableInterface.C

```diff
@@ -155,6 +155,16 @@ MooseVariableInterface::secondTest()
   return const_cast<VariableTestSecond &>(_variable->secondPhi());
 }
 
+VariableTestSecond &
+MooseVariableInterface::secondTestFace()
+{
+  if (_nodal)
+    mooseError("Nodal variables do not have second derivatives");
+
+  return const_cast<VariableTestSecond &>(_variable->secondPhiFace());
```

Comment:
Why don't you make that method return a const reference? We should switch all of these methods over at some time.